plugins/orchestrator-backend/src/service/WorkflowService.ts

```diff
@@ -53,12 +53,12 @@ export class WorkflowService {
   }
 
   async saveWorkflowDefinition(item: WorkflowItem): Promise<WorkflowItem> {
-    const workflowFormat = extractWorkflowFormatFromUri(item.uri);
+    const workflowFormat = extractWorkflowFormatFromUri(item.uri ?? '');
     const definitionsPath = this.resolveResourcePath(
-      `${item.definition.id}.sw.${workflowFormat}`,
+      `${item.definition?.id}.sw.${workflowFormat}`,
```

Comment:
Shouldn't we abort the "save" operation if `uri` or `definition` is `undefined`?